### PR TITLE
migrated transaction.php to use twig

### PR DESCRIPTION
**Pull request by bradymiller**
Proposed conversion of transaction.php to use twig template by Robert.

### CODE REVIEW

**Comment by robertdown** [comment]
Location of the back to patient button doesn't make much sense, am fixing now

File: interface/patient_file/transaction/transactions.php [hunk 1]
```diff
@@ -1,104 +1,44 @@
...
+ *
+ */
+
+
+include_once "../../globals.php";
+include_once "$srcdir/transactions.inc";
+require_once "$srcdir/options.inc.php";
+
+
+$include_standard_style_js = ['common.js'];
```
Comment (bradymiller) [inline interface/patient_file/transaction/transactions.php]: lets put this right above ob_start();, so it's clear where it belongs.

File: interface/patient_file/transaction/transactions.php [hunk 2]
```diff
@@ -1,104 +1,44 @@
...
+
+/*
+ * This code kind of makes me nervous - it forces in the output of the standard
+ * header into a variable so we can pass it to the twig twmplate.
+ *
+ * I'm nervous because I don't have a lot of experience with ob_* functions.
+ * Ideally, once we get the assets helper class up and running we will use that
+ * instead. See https://stackoverflow.com/questions/2830366/require-once-to-variable
+ * for info on this method - RD
```
Comment (bradymiller) [inline interface/patient_file/transaction/transactions.php]: I agree with this assessment.
Why not make a class now that does the stuff below, so it at least looks nicer here. And then can turn that "rough" class into the fancy class when it's ready?

Comment (bradymiller) [inline interface/patient_file/transaction/transactions.php]: And if do make that class, wouldn't it make more sense to make it a function called in twig template?

File: interface/patient_file/transaction/transactions.php [hunk 3]
```diff
@@ -1,104 +1,44 @@
...
-        <a href="add_transaction.php" class="btn btn-default btn-add" onclick="top.restoreSession()">
-            <?php echo xlt('Add'); ?></a>
-        <a href="print_referral.php" onclick="top.restoreSession()" class="btn btn-print btn-default" onclick="top.restoreSession()">
-            <?php echo xlt('View Blank Referral Form'); ?></a>
-    </div>
 
-    <div class='text'>
-        <?php if ($result = getTransByPid($pid)) { ?>
+    $item['refer_date'] = oeFormatShortDate($item['refer_date']);
+    $item['title_field'] = generate_display_field(['data_type' => 1, 'list_id' => 'transactions'], $item['title']);
```
Comment (bradymiller) [inline interface/patient_file/transaction/transactions.php]: note this function returns a value already html escaped.

File: interface/patient_file/transaction/transactions.php [hunk 4]
```diff
@@ -1,104 +1,44 @@
...
+$viewArgs['transactions'] = $result;
+$viewArgs['header'] = $header;
 
-        <?php } else { ?>
-            <span class="text"><?php echo xlt('There are no transactions on file for this patient.'); ?></span>
-        <?php } ?>
-    </div>
-</body>
-</html>
+$GLOBALS['twigLoader']->addPath($GLOBALS['template_dir']);
```
Comment (bradymiller) [inline interface/patient_file/transaction/transactions.php]: this is really neat btw

File: interface/themes/core.css [hunk 5]
```diff
@@ -69,4 +69,8 @@ button[href*="deleter"]:before {
 
 .btn-print::before {
     content: '\f02f\00a0\00a0';
-}
\ No newline at end of file
+}
+
+.btn-back::before {
```
Comment (bradymiller) [inline interface/themes/core.css]: there is my best friend, btn-back

File: interface/patient_file/transaction/transactions.php (hunk 2 above)
Comment (robertdown) [inline interface/patient_file/transaction/transactions.php]: Working on it now, and yes to the autoload function

File: interface/patient_file/transaction/transactions.php (hunk 4 above)
Comment (robertdown) [inline interface/patient_file/transaction/transactions.php]: Thanks! Just wait 'till I abstract it away to a config file :)

Comment (robertdown) [inline interface/patient_file/transaction/transactions.php]: Actually I need to move that to globals

File: templates/patient_file/transactions/list.html.twig [hunk 6]
```diff
@@ -0,0 +1,95 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>{{ 'Patient Transactions'|translate }}</title>
+    {{ header|raw }}
```
Comment (bradymiller) [inline templates/patient_file/transactions/list.html.twig]: see my comment above regarding a function to call the class (that doesn't exist yet) here.

File: templates/patient_file/transactions/list.html.twig [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+                                {% endif %}
+                                {% if row.title == 'LBTref' %}
+                                    <a href='print_referral.php?transid={{ row.id|e('html_attr') }}' onclick='top.restoreSession();'
+                                       class='btn btn-view btn-default'>
+                                        {{ "View"|translate }}
+                                    </a>
+                                {% endif %}
+                            </div>
+                        </td>
+                        <td>{{ row.title_field }}</td>
```
Comment (bradymiller) [inline templates/patient_file/transactions/list.html.twig]: this item gets translated

Comment (bradymiller) [inline templates/patient_file/transactions/list.html.twig]: actually this is more complicated. it needs to be run through the generate_display_field() function.

File: templates/patient_file/transactions/list.html.twig [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+                        <td>{{ row.date }}</td>
```
Comment (bradymiller) [inline templates/patient_file/transactions/list.html.twig]: And this data had quite a bit of manipulation in prior algorithm and also is run through the oeFormatShortDate function.

**Comment by bradymiller** [comment]
For oeFormatShortDate and generate_display_field(), are these more functions that need to be wrapped through twig. If we play this out through entire codebase, aren't we just going to be passing a huge amount of wrapper functions over to twig in the end. Or should we be doing this all in the upstream before bringing in the template?

**Comment by danehrlich1** [comment]
Why was `transactions.php` converted to a Twig Template? I've always heard good things about Twig. Just wanted to ask and this seemed easier than going on the chat :) @robertdown

**Comment by robertdown** [comment]
Long-term goal is to convert all the PHP-based templates to Twig. We will likely have Twig-based templates and React-based templates.

**Pull request closed without merging**